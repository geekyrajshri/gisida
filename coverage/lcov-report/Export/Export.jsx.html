<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Export/Export.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">Export</a> Export.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.89% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>24/27</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">58.06% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>18/31</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>7/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/2</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70</td><td class="line-coverage quiet"><span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="branch-1 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="else path not taken" >E</span><span class="branch-1 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="if path not taken" >I</span><span class="branch-2 cbranch-no" title="branch not covered" ><span class="branch-1 cbranch-no" title="branch not covered" ><span class="branch-0 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="if path not taken" >I</span><span class="missing-if-branch" title="if path not taken" >I</span><span class="branch-0 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="else path not taken" >E</span><span class="missing-if-branch" title="else path not taken" >E</span><span class="missing-if-branch" title="else path not taken" >E</span></span></span></span></span></span></span></span></span></span>
&nbsp;
require('./Export.scss');
&nbsp;
&nbsp;
class Export extends React.Component {
&nbsp;
  static screenshot() {
    return (
      &lt;div id="screenshot-modal" className="modal fade" tabIndex="-1" role="dialog"&gt;
        &lt;div className="modal-dialog" role="document"&gt;
          &lt;div className="modal-content"&gt;
            &lt;button type="button" className="close" data-dismiss="modal" aria-label="Close"&gt;&lt;span aria-hidden="true"&gt;&amp;times;&lt;/span&gt;&lt;/button&gt;
            &lt;div className="field-image" /&gt;
            &lt;div className="modal-body" /&gt;
          &lt;/div&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    );
  }
&nbsp;
  static getMapStatus() {
    const options = {};
    // get layers
    let layers = '';
    const layerState = $('.map-1 input[data-layer]');
    $.each(layerState, (index, layer) =&gt; {
      if ($(layer).prop('checked') === true) {
        layers += `&amp;${$(layer).data('layer')}`;
      }
    });
&nbsp;
    // get lat/lng, zoom
    const center = window.maps[0].getCenter();
    const zoom = window.maps[0].getZoom();
    options.lat = center.lat;
    options.lng = center.lng;
    options.zoom = zoom;
    options.layers = layers;
    return options;
  }
&nbsp;
  static getImageUrl() {
    const options = Export.getMapStatus();
    const url = `https://onacapture.com/export/&amp;${options.lng}&amp;${options.lat}&amp;${options.zoom}${options.layers}`;
    $.get(url)
     .done((data) =&gt; {
       $('#screenshot-modal .field-image').prepend(`&lt;a target="_blank" href="${data.url}"&gt;&lt;img src="${data.url}" /&gt;&lt;/a&gt;`);
     });
  }
&nbsp;
  static showScreenshotModal(e) {
    e.preventDefault();
    Export.getImageUrl();
    $('#screenshot-modal').modal('show');
  }
&nbsp;
  render() {
    return (
      &lt;div&gt;
        &lt;a className="export-btn" href="#" onClick={e =&gt; Export.showScreenshotModal(e)}&gt;&lt;span className="glyphicon glyphicon-camera" /&gt;&lt;/a&gt;
        { Export.screenshot() }
      &lt;/div&gt;
    );
  }
&nbsp;
}
&nbsp;
export default Export;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jun 27 2017 15:27:56 GMT+0300 (EAT)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
