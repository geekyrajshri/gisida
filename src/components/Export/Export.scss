.export-btn {
  width: 34px;
  background: #fff;
  z-index: 500;
  right: 8px;
  top: 150px;
  display: block;
  position: absolute;
  font-size: 18px;
  text-align: center;
  line-height: 1.8;
  color: #222;
  border: 2px solid #cdcdcd;
  border-radius: 7px;
}

.export-btn-map-2 {
  
}

.modal-content {
  padding: 40px;
  
  .close {
    position: absolute;
    right: 15px;
    top: 15px;
    span {
      font-size: 30px;
    }
  }
  
  .field-image img {
    width: 100%;
    height: auto;
    min-height: 350px;
    background: #eee url(/assets/img/loader.svg) no-repeat center center;
    border: 1px solid #ccc;
    margin-bottom: 20px;
  }
}
