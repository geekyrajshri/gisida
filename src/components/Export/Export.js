require('./Export.scss')
import React from 'react'
import { Component } from 'react'

class Export extends Component {
  constructor(props) {
    super(props)

  }
  
  componentDidMount() {
  
  }
  
  getMapStatus() {
    var options = {}
    // get layers
    
    var layers = '';
    var layerState = $('.map-1 input[data-layer]')

    $.each(layerState, function(index, layer) {
      if ($(layer).prop('checked') === true) {
        layers += '&' + $(layer).data('layer')
      }
    })
    console.log(layers)
   
    // get lat/lng, zoom  
    var center = window.maps[0].getCenter()
    var zoom = window.maps[0].getZoom()
      options.lat = center.lat,
      options.lng = center.lng,
      options.zoom = zoom,
      options.layers = layers
    return options 
  }
  
  
  getImageUrl() {
    var options = this.getMapStatus()
    console.log(options) 
    var url = 'https://onacapture.com/export/&' + options.lng + '&' + options.lat + '&' + options.zoom + options.layers;
    console.log(url)
    $.get(url)
     .done(function(data) {
    
      console.log(data)
      $('#screenshot-modal .field-image').prepend('<a target="_blank" href="' + data.url + '"><img src="' +  data.url + '" /></a>');
    }) 
  }
  
  
 showScreenshotModal(e) {
  e.preventDefault()
   this.getImageUrl()
   $('#screenshot-modal').modal('show')
 }
  
  
 screenshot() {
   return(
     <div id="screenshot-modal" className="modal fade" tabIndex="-1" role="dialog">
      <div className="modal-dialog" role="document">
       <div className="modal-content">
         <button type="button" className="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
          <div className="field-image">
           
          </div>
         <div className="modal-body">
           
         </div>
       </div>
      </div>
    </div>
   )
 } 
  
 
  render() {
    return(
      <div>
        <a className="export-btn" href="#" onClick={(e) => this.showScreenshotModal(e)}><span className="glyphicon glyphicon-camera"></span></a>
        { this.screenshot() }
      </div>
    )
   }
    
}

export default Export