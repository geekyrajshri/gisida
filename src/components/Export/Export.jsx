import React from 'react';

require('./Export.scss');

class Export extends React.Component {
  static screenshot() {
    return (
      <div id="screenshot-modal" className="modal fade" tabIndex="-1" role="dialog">
        <div className="modal-dialog" role="document">
          <div className="modal-content">
            <button type="button" className="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
            <div className="field-image" />
            <div className="modal-body" />
          </div>
        </div>
      </div>
    );
  }

  static getMapStatus() {
    const options = {};
    // get layers
    let layers = '';
    const layerState = $('.map-1 input[data-layer]');
    $.each(layerState, (index, layer) => {
      if ($(layer).prop('checked') === true) {
        layers += `&${$(layer).data('layer')}`;
      }
    });
    // get lat/lng, zoom
    const center = window.maps[0].getCenter();
    const zoom = window.maps[0].getZoom();
    options.lat = center.lat;
    options.lng = center.lng;
    options.zoom = zoom;
    options.layers = layers;

    return options;
  }

  static getImageUrl() {
    const options = Export.getMapStatus();
    const url = `https://onacapture.com/export/&${options.lng}&${options.lat}&${options.zoom}${options.layers}`;
    $.get(url)
      .done((data) => {
        $('#screenshot-modal .field-image').prepend(`<a target="_blank" href="${data.url}"><img src="${data.url}" /></a>`);
      });
  }

  static showScreenshotModal(e) {
    e.preventDefault();
    Export.getImageUrl();
    $('#screenshot-modal').modal('show');
  }

  componentDidMount() {
    // place the btns in their map divs
    // @todo: tidy this up
    const btn1 = $('.export-btn-map-1');
    $('#map-1').append(btn1);
    const btn2 = $('.export-btn-map-2');
    $('#map-2').append(btn2);
  }

  render() {
    return (
      <div>
        <a className={`export-btn export-btn-${this.props.map}`} href="#" onClick={e => Export.showScreenshotModal(e)}><span className="glyphicon glyphicon-camera" /></a>
        {Export.screenshot()}
      </div>
    );
  }

}

export default Export;

Export.propTypes = {
  map: React.PropTypes.string.isRequired,
};

Export.defaultProps = {
  map: '',
};

