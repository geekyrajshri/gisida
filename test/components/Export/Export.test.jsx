import React from 'react';
import renderer from 'react-test-renderer';
import Export from '../../../src/components/Export/Export';

describe('Export', () => {
    it('Export renders export icon and screen', () => {
        const component = renderer.create(<Export map={'map-1'} />);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});
