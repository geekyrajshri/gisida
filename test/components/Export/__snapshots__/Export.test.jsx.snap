// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Export Export renders export icon and screen 1`] = `
<div>
  <a
    className="export-btn export-btn-"
    href="#"
    onClick={[Function]}
  >
    <span
      className="glyphicon glyphicon-camera"
    />
  </a>
  <div
    className="modal fade"
    id="screenshot-modal"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
        <div
          className="field-image"
        />
        <div
          className="modal-body"
        />
      </div>
    </div>
  </div>
</div>
`;
